InAppUtils/InAppUtils.m

```diff
@@ -6,243 +6,244 @@
-                RCTResponseSenderBlock callback = _callbacks[key];
-                if (callback) {
-                    if (transaction.error.code == SKErrorPaymentCancelled) {
-                        callback(@[@"user_cancelled"]);
-                        [_callbacks removeObjectForKey:key];
-                    }
-                    callback(@[RCTJSErrorFromNSError(transaction.error)]);
-                    [_callbacks removeObjectForKey:key];
-                } else {
-                    RCTLogWarn(@"No callback registered for transaction with state failed.");
-                }
-                [[SKPaymentQueue defaultQueue] finishTransaction:transaction];
-                break;
-            }
-            case SKPaymentTransactionStatePurchased: {
-                NSString *key = RCTKeyForInstance(transaction.payment.productIdentifier);
-                RCTResponseSenderBlock callback = _callbacks[key];
-                if (callback) {
-                    NSDictionary *purchase = @{
-                                              @"transactionDate": @(transaction.transactionDate.timeIntervalSince1970 * 1000),
-                                              @"transactionIdentifier": transaction.transactionIdentifier,
-                                              @"productIdentifier": transaction.payment.productIdentifier,
-                                              @"transactionReceipt": [[transaction transactionReceipt] base64EncodedStringWithOptions:0]
-                                              };
-                    callback(@[[NSNull null], purchase]);
-                    [_callbacks removeObjectForKey:key];
-                } else {
-                    RCTLogWarn(@"No callback registered for transaction with state purchased.");
-                }
-                [[SKPaymentQueue defaultQueue] finishTransaction:transaction];
-                break;
-            }
-            case SKPaymentTransactionStateRestored:
-                [[SKPaymentQueue defaultQueue] finishTransaction:transaction];
-                break;
-            case SKPaymentTransactionStatePurchasing:
-                NSLog(@"purchasing");
-                break;
-            case SKPaymentTransactionStateDeferred:
-                NSLog(@"deferred");
-                break;
-            default:
-                break;
-        }
-    }
+   for (SKPaymentTransaction *transaction in transactions) {
+       switch (transaction.transactionState) {
+           case SKPaymentTransactionStateFailed: {
+               NSString *key = RCTKeyForInstance(transaction.payment.productIdentifier);
+               RCTResponseSenderBlock callback = _callbacks[key];
+               if (callback) {
+                   NSLog(@"Call back from updatedTransactions: %@", transaction.error);
+                   callback(@[translateErrorCode(transaction.error.code, @"restore_failed")]);
```

Comment:
Similar to this? We would send it the code to switch against and the key we should be returning?